modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2606,6 +2629,7 @@ private CompletableFuture<ReplicaResult> processSingleEntryAction(ReadWriteSingl
 
                     return takeLocksForDelete(row, rowId, txId)
                             .thenCompose(ignored -> validateOperationAgainstSchema(request.transactionId()))
+                            .thenCompose(catalogVersion -> checkCleanup(rowId, catalogVersion))
```

Comment:
Seems that validateOperationAgainstSchema and checkCleanup can be awaited in parallel, right? If true I'd rather consider following approach
```
f = validateOperationAgainstSchema(request.transactionId())
allOf(f, checkCleanup).thenCompose(ignored -> {catalogVersion = f.join())
```
It will also allow to remove T result param from the checkCleanup()

Reply:
> Is there any benefit from doing this?
Minor and in rare cases whe operation tiriggers the cleanup itself.

Feel free to use your original solution.